lib/protokoll/formater.rb

```diff
@@ -61,12 +62,13 @@ def extract_sufix(pattern)
     end
 
     def expand_times(pattern)
-      pattern.sub("%y", Time.now.strftime("%y"))
-        .sub("%Y", Time.now.strftime("%Y"))
-        .sub("%d", Time.now.strftime("%d"))
-        .sub("%m", Time.now.strftime("%m"))
-        .sub("%M", Time.now.strftime("%M"))
-        .sub("%H",  Time.now.strftime("%H"))
+      pat = pattern.dup # pattern is a frozen string.
+      pat.sub!("%y", Time.now.strftime("%y"))
```

Comment:
Using `sub!` again to avoid multiple string allocations. Every time `sub` is called, a new string is allocated.